cmd/av/stack_reparent.go

```diff
@@ -1,24 +1,281 @@
+			repo:    repo,
+			db:      db,
+			spinner: spinner.New(spinner.WithSpinner(spinner.Dot)),
+		}, opts...)
+		model, err := p.Run()
+		if err != nil {
+			return err
+		}
+		if err := model.(stackReparentViewModel).err; err != nil {
+			return actions.ErrExitSilently{ExitCode: 1}
+		}
+		if s := model.(stackReparentViewModel).rebaseConflictErrorHeadline; s != "" {
+			return actions.ErrExitSilently{ExitCode: 1}
+		}
 		return nil
 	},
 }
+
+type stackReparentState struct {
+	InitialBranch   string
+	NewParentBranch string
+	Seq             *sequencer.Sequencer
+}
+
+type stackReparentSeqResult struct {
+	result *git.RebaseResult
+	err    error
+}
+
+type stackReparentViewModel struct {
+	repo    *git.Repo
+	db      meta.DB
+	state   *stackReparentState
+	spinner spinner.Model
+
+	rebaseConflictErrorHeadline string
+	rebaseConflictHint          string
+	abortedBranch               plumbing.ReferenceName
+	err                         error
+}
+
+func (vm stackReparentViewModel) Init() tea.Cmd {
+	return tea.Batch(vm.spinner.Tick, vm.initCmd)
+}
+
+func (vm stackReparentViewModel) Update(msg tea.Msg) (tea.Model, tea.Cmd) {
+	switch msg := msg.(type) {
+	case error:
+		vm.err = msg
+		return vm, tea.Quit
+	case *stackReparentState:
+		vm.state = msg
+		if stackReparentFlags.Skip || stackReparentFlags.Continue || stackReparentFlags.Abort {
+			if stackReparentFlags.Abort {
+				vm.abortedBranch = vm.state.Seq.CurrentSyncRef
+			}
+			return vm, vm.runSeqWithContinuationFlags
+		}
+		return vm, vm.runSeq
+	case *stackReparentSeqResult:
+		if msg.err == nil && msg.result == nil {
+			// Finished the sequence.
+			if err := vm.repo.WriteStateFile(git.StateFileKindReparent, nil); err != nil {
+				vm.err = err
+			}
+			if _, err := vm.repo.CheckoutBranch(&git.CheckoutBranch{Name: vm.state.InitialBranch}); err != nil {
+				vm.err = err
+			}
+			return vm, tea.Quit
+		}
+		if msg.result != nil && msg.result.Status == git.RebaseConflict {
+			vm.rebaseConflictErrorHeadline = msg.result.ErrorHeadline
+			vm.rebaseConflictHint = msg.result.Hint
+			if err := vm.repo.WriteStateFile(git.StateFileKindReparent, vm.state); err != nil {
+				vm.err = err
+			}
+			return vm, tea.Quit
+		}
+		vm.err = msg.err
+		if vm.err != nil {
+			return vm, tea.Quit
+		}
+		return vm, vm.runSeq
+	case spinner.TickMsg:
+		var cmd tea.Cmd
+		vm.spinner, cmd = vm.spinner.Update(msg)
+		return vm, cmd
+	case tea.KeyMsg:
+		switch msg.String() {
+		case "ctrl+c":
+			return vm, tea.Quit
+		}
+	}
+	return vm, nil
+}
+
+func (vm stackReparentViewModel) View() string {
+	sb := strings.Builder{}
+	if vm.state != nil && vm.state.Seq != nil {
+		sb.WriteString("Reparenting " + vm.state.InitialBranch + " onto " + vm.state.NewParentBranch + "...\n")
+		if vm.state.Seq.CurrentSyncRef != "" {
+			sb.WriteString("Restacking " + vm.state.Seq.CurrentSyncRef.Short() + "...\n")
+		} else if vm.abortedBranch != "" {
+			sb.WriteString("Restack aborted\n")
```

Comment:
if the action is aborted would we want to return here early?

Reply:
No. We want to show the tree still.